src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -504,210 +560,333 @@ core::smart_refctd_ptr<const CSPIRVIntrospector::CIntrospectionData> CSPIRVIntro
+            res->combinedImageSampler.viewType = spvcrossImageType2ImageView(type.image);
+            res->combinedImageSampler.shadow = type.image.depth;
+            res->combinedImageSampler.multisample = type.image.ms;
         }
     }
-    for (const spirv_cross::Resource& r : resources.separate_images)
-    {
-        SShaderResourceVariant& res = addResource_common(r, ESRT_SAMPLED_IMAGE, mapId2SpecConst);
-    }
-    for (const spirv_cross::Resource& r : resources.separate_samplers)
-    {
-        SShaderResourceVariant& res = addResource_common(r, ESRT_SAMPLER, mapId2SpecConst);
-    }
-    for (auto& descSet : introData->descriptorSetBindings)
-        std::sort(descSet.begin(), descSet.end(), [](const SShaderResourceVariant& _lhs, const SShaderResourceVariant& _rhs) { return _lhs.binding < _rhs.binding; });
+    // skip cause we don't support
+    //for (const spirv_cross::Resource& r : resources.separate_images) {}
+    //for (const spirv_cross::Resource& r : resources.separate_samplers) {}
+    for (auto& descSet : stageIntroData->m_descriptorSetBindings)
+        std::sort(descSet.begin(), descSet.end());
 
+    auto getStageIOtype = [&comp](CSPIRVIntrospector::CStageIntrospectionData::SInterface& glslType, uint32_t id, uint32_t base_type_id)
+        {
+            const auto& type = comp.get_type(base_type_id);
+            glslType.baseType = spvcrossType2E_TYPE(type.basetype);
+            glslType.elements = type.vecsize;
+            glslType.location = comp.get_decoration(id, spv::DecorationLocation);
 
-    auto getStageIOtype = [&_comp](uint32_t _base_type_id)
-    {
-        const auto& type = _comp.get_type(_base_type_id);
-        decltype(SShaderInfoVariant::glslType) glslType;
-        glslType.basetype = spvcrossType2E_TYPE(type.basetype);
-        glslType.elements = type.vecsize;
-
-        return glslType;
-    };
+            return glslType;
+        };
 
     // in/out
     for (const spirv_cross::Resource& r : resources.stage_inputs)
     {
-        SShaderInfoVariant& res = addInfo_common(r, ESIT_STAGE_INPUT);
-        res.glslType = getStageIOtype(r.base_type_id);
+                                                                                            // TODO: hash map instead
+        CSPIRVIntrospector::CStageIntrospectionData::SInputInterface& res = stageIntroData->m_input.emplace_back();
+        getStageIOtype(res, r.id, r.base_type_id);
     }
+
+    if (shaderStage == IShader::ESS_FRAGMENT)
+        stageIntroData->m_output = FragmentOutputVecT();
+    else
+        stageIntroData->m_output = OutputVecT();
+
     for (const spirv_cross::Resource& r : resources.stage_outputs)
     {
-        SShaderInfoVariant& res = addInfo_common(r, ESIT_STAGE_OUTPUT);
-        res.glslType = getStageIOtype(r.base_type_id);
+        if (shaderStage == IShader::ESS_FRAGMENT)
+        {
+            // TODO: desn't path here for frag_test1.hlsl, why?
```

Comment:
someone didn't figure out  the shader stage? or is a stale TODO?

Reply:
stale